platform/persistence/platform-persistence-core-impl/src/main/java/org/codice/ddf/persistence/internal/PersistentStoreImpl.java

```diff
@@ -131,13 +135,32 @@ private void doRollback(SolrClient solrClient, String type) {
   }
 
   @Override
-  // Returned Map will have suffixes in the key names - client is responsible for handling them
+  /**
+   * {@inheritDoc} Returned Map will have suffixes in the key names - client is responsible for
+   * handling them
+   */
   public List<Map<String, Object>> get(String type, String cql) throws PersistenceException {
+    return get(type, cql, DEFAULT_START_INDEX, DEFAULT_PAGE_SIZE);
+  }
+
+  @Override
+  public List<Map<String, Object>> get(String type, String cql, int startIndex, int pageSize)
+      throws PersistenceException {
     if (StringUtils.isBlank(type)) {
       throw new PersistenceException(
           "The type of object(s) to retrieve must be non-null and not blank, e.g., notification, metacard, etc.");
     }
 
+    if (startIndex < 0) {
+      throw new IllegalArgumentException("The start index must be nonnegative.");
+    }
+
+    if (pageSize <= 0 || pageSize > MAX_PAGE_SIZE) {
```

Comment:
❓ Do we have to enforce a max page size? 

😞 I really wish we didn't have to have a magic internal page size. I'd rather us just make sure people are using it correctly rather than enforcing a magic size, since people will just end up having to create something like the ResultIterator, and then get all the results and hold them in memory anyways.

Reply:
It could be a security threat if that parameter is eventually exposed through an endpoint.  That was the case for the catalog query.

@codice/security

Reply:
Should we then just cap the query at 1000 then instead of throwing an illegal arg exception , similiar to what catalog framework does now?

Reply:
Also that seems like we should be preventing the endpoints from querying > 1000 instead of crippling the entire product D:

Reply:
I think throwing an exception is fine. It ensures the calling code is correct. Granted, it's a stretch but could be considered an "API change", but at the end of the day anyone calling the persistence store should be doing it properly so there is no ambiguity about the result set.